app/src/main/java/org/simple/clinic/teleconsultlog/prescription/patientinfo/TeleconsultPatientInfoView.kt

```diff
@@ -0,0 +1,122 @@
+package org.simple.clinic.teleconsultlog.prescription.patientinfo
+
+import android.content.Context
+import android.os.Parcelable
+import android.util.AttributeSet
+import androidx.cardview.widget.CardView
+import io.reactivex.Observable
+import kotlinx.android.synthetic.main.view_teleconsult_patient_info.view.*
+import org.simple.clinic.R
+import org.simple.clinic.di.injector
+import org.simple.clinic.mobius.MobiusDelegate
+import org.simple.clinic.patient.DateOfBirth
+import org.simple.clinic.patient.Patient
+import org.simple.clinic.patient.PatientAddress
+import org.simple.clinic.patient.PatientProfile
+import org.simple.clinic.patient.displayLetterRes
+import org.simple.clinic.router.screen.ScreenRouter
+import org.simple.clinic.teleconsultlog.prescription.TeleconsultPrescriptionScreenKey
+import org.simple.clinic.util.UserClock
+import org.simple.clinic.util.unsafeLazy
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import javax.inject.Inject
+import javax.inject.Named
+
+class TeleconsultPatientInfoView constructor(
+    context: Context,
+    attrs: AttributeSet?
+) : CardView(context, attrs), TeleconsultPatientInfoUi {
+
+  @Inject
+  lateinit var userClock: UserClock
+
+  @Inject
+  @Named("full_date")
+  lateinit var dateFormatter: DateTimeFormatter
+
+  @Inject
+  lateinit var effectHandler: TeleconsultPatientInfoEffectHandler
+
+  @Inject
+  lateinit var screenRouter: ScreenRouter
+
+  private val delegate by unsafeLazy {
+    val screenKey = screenRouter.key<TeleconsultPrescriptionScreenKey>(this)
```

Comment:
Not necessary to change this, but something to keep in mind here that we are coupling this view to the `TeleconsultPrescriptionScreen`.

Reply:
I am assuming that this is because this view is going to be embedded within that screen.

Reply:
Yes. Is there a better way of handling these cases?

Reply:
Not at the moment, I think it's find to keep this as is. A more scalable approach might be to have more find grained Dependency scopes which we can use to inject these properties into sub views.

Reply:
Agreed ✅